front/components/vaults/VaultManagedDatasourcesModal.tsx

```diff
@@ -0,0 +1,219 @@
+import type {
+  ConnectorProvider,
+  DataSourceViewType,
+  VaultSelectedDataSources,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { useState } from "react";
+
+import DataSourceResourceSelectorTree from "@app/components/DataSourceResourceSelectorTree";
+import { useParentResourcesById } from "@app/hooks/useParentResourcesById";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+
+export default function VaultManagedDataSourcesModal({
+  isOpen,
+  setOpen,
+  owner,
+  systemVaultDataSourceViews,
+  onSave,
+  initialSelectedDataSources,
+}: {
+  isOpen: boolean;
+  setOpen: (isOpen: boolean) => void;
+  owner: WorkspaceType;
+  systemVaultDataSourceViews: DataSourceViewType[];
+  onSave: (dsConfigs: VaultSelectedDataSources) => void;
+  initialSelectedDataSources: DataSourceViewType[];
+}) {
+  const [selectedDataSources, setSelectedDataSources] =
+    useState<VaultSelectedDataSources>(
+      initialSelectedDataSources.map((ds) => ({
+        name: ds.name,
+        parentsIn: ds.parentsIn ?? null,
+      }))
+    );
+  const [hasChanged, setHasChanged] = useState(false);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => {
+        setOpen(false);
+      }}
+      onSave={() => {
+        onSave(selectedDataSources);
+        setOpen(false);
+      }}
+      hasChanged={hasChanged}
+      variant="side-md"
+      title="Add connected datasources"
+    >
+      <div className="w-full pt-12">
+        <div className="overflow-x-auto">
+          <Tree isLoading={false}>
+            {systemVaultDataSourceViews.map((dataSourceView) => {
+              return (
+                <VaultManagedDataSourceTree
+                  key={dataSourceView.name}
+                  owner={owner}
+                  dataSourceView={dataSourceView}
+                  selectedDataSources={selectedDataSources}
+                  setSelectedDataSources={setSelectedDataSources}
+                  hasChanged={hasChanged}
+                  setHasChanged={setHasChanged}
+                />
+              );
+            })}
+          </Tree>
+        </div>
+      </div>
+    </Modal>
+  );
+}
+
+function VaultManagedDataSourceTree({
+  owner,
+  dataSourceView,
+  selectedDataSources,
+  setSelectedDataSources,
+  hasChanged,
+  setHasChanged,
+}: {
+  owner: WorkspaceType;
+  dataSourceView: DataSourceViewType;
+  selectedDataSources: VaultSelectedDataSources;
+  setSelectedDataSources: (
+    prevState: (prevState: VaultSelectedDataSources) => VaultSelectedDataSources
+  ) => void;
+  hasChanged: boolean;
+  setHasChanged: (hasChanged: boolean) => void;
+}) {
+  const config =
+    CONNECTOR_CONFIGURATIONS[
+      dataSourceView.connectorProvider as ConnectorProvider
+    ];
+  const LogoComponent = config?.logoComponent ?? null;
+  const selectedDs = selectedDataSources.find(
+    (ds) => ds.name === dataSourceView.name
+  );
+  const { parentsById, setParentsById } = useParentResourcesById({
```

Comment:
Ideally this code path should go through the data_source_views endpoint and not data_sources.